device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/FileUpload.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
Most of this is copied from the existing implementation to a new file that isn't in its own directory. With this move, I made the constructors and methods package private so that users don't see them.